### PR TITLE
Fix bug with mavenSettings in  WrapperRunner while running tests locally

### PR DESCRIPTION
**Pull request by ia3andy**
```
[ERROR] Error executing Maven.
[ERROR] The specified user settings file does not exist: /Users/ia3andy/.m2/settings.xml
```

### CODE REVIEW

**Comment by aloubyansky** [comment]
What kind of use-case is this fixing?

**Comment by aloubyansky** [comment]
I suppose there could be a relative path evaluation that could go wrong. But just to clarify.

**Comment by gastaldi** [comment]
> I suppose there could be a relative path evaluation that could go wrong. But just to clarify.

IMHO if the `-s` flag was explicitly set to a non-existent file it should fail, so it would give the dev a hint of what's wrong. Silently ignoring it in this case is bad.

**Comment by aloubyansky** [comment]
If it's a relative path though, it could be resolved correctly against the current project but the one it is passed to. So, perhaps, here we should be making sure we are always propagating an absolute path. But again it would be good to clarify what issue Andy ran into.

**Comment by ia3andy** [comment]
maven is working fine locally without this file.. this bug was introduced when we started reading `maven.settings` in WrapperRunner.

**Comment by ia3andy** [comment]
@gastaldi  I don't think we should propagate the `-s` with a non existing file.

**Comment by ia3andy** [comment]
Here is when it started: https://github.com/quarkusio/quarkus/pull/21673

**Comment by ia3andy** [comment]
@gastaldi no it's using the properties, I am not setting it manually.

**Comment by gastaldi** [comment]
> @gastaldi I don't think we should propagate the `-s` with a non existing file.

We should propagate what's been set in the original command. If it's an invalid path, that's the user's fault, the error should not be swallowed

**Comment by gastaldi** [comment]
So I guess the fix is to make sure that `getMavenSettingsArg()` returns a value ONLY if it's set

**Comment by ia3andy** [comment]
> So I guess the fix is to make sure that `getMavenSettingsArg()` returns a value ONLY if it's set. Checking if the file exists here is bad because it may lead to unexpected behavior when the command executes succesfully with a non-existent pathII

I don't think we can can know if `-s` is set or not. 

Maven is failing only when explicitly using `-s` with a non existent file. Here this is propagation, we shouldn't propagate something which makes the runner fail.

So maybe it's not the best solution, but it's better than the current code..

**Comment by ia3andy** [comment]
@gastaldi if you have a better solution to not propagate `-s` when you should not, let me know

**Comment by gastaldi** [comment]
> Maven is failing only when explicitly using `-s` with a non existent file. Here this is propagation, we shouldn't propagate something which makes the runner fail.

It's okay to fail if the parameters are wrong, I think the bug is that -s is specified here but not in the original command

**Comment by aloubyansky** [comment]
We check whether `maven.settings` is set first. If it is we can check whether it's a valid path. If it's not, we should return null.
If `maven.settings` isn't set we can check whether `-s` was provided. If it was we should propagate it as an absolute path.

**Comment by aloubyansky** [comment]
We should always propagate it with `-s` because `maven.settings` is a Quarkus only property.

File: independent-projects/tools/devtools-testing/src/main/java/io/quarkus/devtools/testing/WrapperRunner.java [hunk 1]
```diff
@@ -104,7 +106,7 @@ public static int run(Path projectDir, Wrapper wrapper) {
 
     private static String getMavenSettingsArg() {
         final String mavenSettings = System.getProperty("maven.settings");
-        if (mavenSettings != null) {
+        if (mavenSettings != null && Files.exists(Paths.get(mavenSettings))) {
             return mavenSettings;
         }
```
Comment (gastaldi) [inline independent-projects/tools/devtools-testing/src/main/java/io/quarkus/devtools/testing/WrapperRunner.java:111]: According to @aloubyansky it should be like:

```suggestion
        if (mavenSettings != null) {
            return Files.exists(Paths.get(mavenSettings)) ? mavenSettings : null;
        }
```

Comment (ia3andy) [reply independent-projects/tools/devtools-testing/src/main/java/io/quarkus/devtools/testing/WrapperRunner.java:111]: @aloubyansky shouldn't we use `-s` if the file does not exist?

**Review by aloubyansky — APPROVED** [review]

File: independent-projects/tools/devtools-testing/src/main/java/io/quarkus/devtools/testing/WrapperRunner.java [hunk 2]
```diff
@@ -105,7 +107,7 @@ public static int run(Path projectDir, Wrapper wrapper) {
     private static String getMavenSettingsArg() {
         final String mavenSettings = System.getProperty("maven.settings");
         if (mavenSettings != null) {
-            return mavenSettings;
+            return Files.exists(Paths.get(mavenSettings)) ? mavenSettings : null;
```
Comment (gastaldi) [inline independent-projects/tools/devtools-testing/src/main/java/io/quarkus/devtools/testing/WrapperRunner.java:110]: Minor nitpick: use `Path.of` instead :)

**Review by gastaldi — APPROVED** [review]

File: independent-projects/tools/devtools-testing/src/main/java/io/quarkus/devtools/testing/WrapperRunner.java [hunk 3]
```diff
@@ -105,7 +107,7 @@ public static int run(Path projectDir, Wrapper wrapper) {
...
         }
         return BootstrapMavenOptions.newInstance().getOptionValue(BootstrapMavenOptions.ALTERNATE_USER_SETTINGS);
```
Comment (gastaldi) [inline independent-projects/tools/devtools-testing/src/main/java/io/quarkus/devtools/testing/WrapperRunner.java:112]: BTW shouldn't we make this result an absolute path?

Comment (aloubyansky) [reply independent-projects/tools/devtools-testing/src/main/java/io/quarkus/devtools/testing/WrapperRunner.java:112]: It's worth making sure the ALTERNATE_USER_SETTINGS is an absolute path.

File: independent-projects/tools/devtools-testing/src/main/java/io/quarkus/devtools/testing/WrapperRunner.java (hunk 2 above)
Comment (ia3andy) [reply independent-projects/tools/devtools-testing/src/main/java/io/quarkus/devtools/testing/WrapperRunner.java:110]: Yeah it's a shortcut you are right. I think we can live with it :)

**Comment by ia3andy** [comment]
Error seems unrelated. @aloubyansky should we merge?

**Pull request merged**